Shoko.Server/ImageDownload/ImageDownloadRequest.cs

```diff
@@ -1,17 +1,129 @@
-﻿using Shoko.Models.Enums;
+﻿using System.Net;
+using System.IO;
+using Shoko.Models.Enums;
+using Shoko.Models.Server;
+using Shoko.Server.Extensions;
+using Shoko.Server.Models;
+using Shoko.Server.Server;
+using Shoko.Commons.Utils;
+using System.Threading;
+using System.Net.Http;
 
+#nullable enable
 namespace Shoko.Server.ImageDownload;
 
 public class ImageDownloadRequest
 {
-    public ImageEntityType ImageType { get; set; }
-    public object ImageData { get; set; }
-    public bool ForceDownload { get; set; }
 
-    public ImageDownloadRequest(ImageEntityType imageType, object data, bool forceDownload)
+    private object ImageData { get; }
+
+    public bool ForceDownload { get; }
```

Comment:
We can probably remove this since it's always the same as the `ForceDownload` option in the command(s).

Reply:
Will update the code when i get back home.